src/sagas.js

```diff
@@ -753,10 +780,23 @@ function mergeStorageDomains (storageDomainsInternal) {
 }
 
 function* fetchAllClusters (action) {
+  if (Selectors.getFilter()) {
+    action.payload.additional = ['permissions.role']
+  }
   const clusters = yield callExternalAction('getAllClusters', Api.getAllClusters, action)
 
   if (clusters && clusters['cluster']) {
-    const clustersInternal = clusters.cluster.map(cluster => Api.clusterToInternal({ cluster }))
+    let clustersInternal = []
+    if (!Selectors.getFilter()) {
+      clustersInternal = yield all(clusters.cluster.map(cluster => call(function* () {
+        const permissions = yield callExternalAction('getClusterPermissions', Api.getClusterPermissions, { payload: { clusterId: cluster.id } })
```

Comment:
No, because `Filter: true` return only actual user permissions, `Filter: false` return all permissions that user can see (in case of admin this is all permissions). Also there is no way to get list of group members through RestAPI.
So for admin VM Portal send separate permission request with `Filter: true` to get only actual user permissions.

Reply:
> Also there is no way to get list of group members through RestAPI.

True.  But, you can list the groups a user is a member of, and that is good enough.

At login time, fetch the user's groups `.../api/users/${userId}/groups`, and push them in the config reducer.  Then when you fetch the clusters (or any other thing that needs to be filtered based on permissions), if `Filter: false` you can match the roles on userId and groupId.

Otherwise we'll be stuck with having to fetch the permissions for each object every time a admin user logs in.  That could be a lot of extra fetches.

Reply:
Nice idea!